lib/iris/_cube_coord_common.py

```diff
@@ -28,6 +28,27 @@
 import iris.std_names
 
 
+def get_valid_standard_name(name):
+    # Standard names are optionally followed by a standard name
+    # modifier, separated by one or more blank spaces
+    name_parts = name.split(' ')
+    std_name, std_name_extension = name_parts[0], name_parts[1:]
+
+    error_msg = '{!r} is not a valid standard_name'.format(name)
+
+    if std_name in iris.std_names.STD_NAMES:
+        if std_name_extension:
+            standard_name_modifier = std_name_extension[-1]
+            # Check only blank spaces before valid standard_name_modifier
+            if (any(item for item in std_name_extension[:-1]) or
+                    standard_name_modifier not in
+                    iris.fileformats.cf.STD_NAME_MODIFIERS):
+                raise ValueError(error_msg)
+        return name
```

Comment:
Yeah I agree, it would be nicer.I just couldn't quite think through how to do that, but I will give it another look!

Reply:
@lbdreyer I think that all you need to do is simply move the `return name` outside the `if ... else ...`, as your not changing the contents of `name`, the output of the function is just the same as the input.

If the input `std_name` (from `name`) is valid, then it will pass through the `if ...`, and then if the `standard_name_modifier` is valid it will simply drop through and `return name`. On the otherhand, if the modifier is not valid an exception is raised and the `return name` is never be reached.

If the input `name` is invalid, then it will pass through to the `else ...` where an exception is raised and the `return name` will never be reached.

Should just work, right? Check my logic is right, though...